File: src/main/java/graveler/action/UpdateNeighborStress.java
package graveler.action;

import graveler.util.WorldExtensions;
import net.minecraft.core.BlockPos;
import net.minecraft.core.Direction;

public record UpdateNeighborStress(BlockPos pos) implements Action {
  private static final Direction[] UPDATE_ORDER = {
    Direction.WEST,
    Direction.EAST,
    Direction.NORTH,
    Direction.SOUTH,
    Direction.DOWN,
    Direction.UP
  };

  @Override
  public double weight() {
    return 0.0;
  }

  @Override
  public void apply(ActionContext context) {
    for (Direction direction : UPDATE_ORDER) {
      BlockPos neighbor = pos.relative(direction);
      context.scheduler().schedule(new UpdateStress(neighbor));
    }
  }
}

File: src/main/kotlin/graveler/action/UpdateNeighborStress.kt
package graveler.action

import graveler.util.pointedAt
import net.minecraft.util.Direction
import net.minecraft.util.math.BlockPos

data class UpdateNeighborStress(val pos: BlockPos) : Action {
  override val weight: Double = 0.0

  override fun apply(context: ActionContext) {
    val world = context.world

    UPDATE_ORDER
      .map { pos.offset(it) }
      .map { world.pointedAt(it) }
      .forEach { context.scheduler.schedule(UpdateStress(it.pos)) }
  }

  companion object {
    private val UPDATE_ORDER = arrayOf(Direction.WEST, Direction.EAST, Direction.NORTH, Direction.SOUTH, Direction.DOWN, Direction.UP)
  }
}
